solutions/usaco-939.mdx

```diff
@@ -7,10 +7,81 @@ author: Maggie Liu
 
 [Official Analysis](http://www.usaco.org/current/data/sol_buckets_bronze_open19.html)
 
+## Explanation
+
+If there was no rock, then the number of cows needed would just be the
+horizontal distance between the barn and the lake, plus the vertical distance
+between the barn and the lake, minus $1$. Adding the rock onto the farm usually
+doesn't change the answer. The only case where the rock matters is if the rock,
+barn and lake are on the same line, and the rock is between the barn and the
+lake. In this case, there will need to be $2$ extra cows to go around the rock.
+
 ## Implementation
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
```

Comment:
i thought you were the type to use bits
![image](https://user-images.githubusercontent.com/55369003/124392220-36afb280-dca9-11eb-89cc-50d96ea12f4c.png)